### DISCUSSION TITLE
SedOneStepSimulation

### DISCUSSION THREAD

**Discussion by jmdetloff** [biosimulations/biosimulations#4363] [Q&A] [answered]
A quick question on something I noticed while working on updating the Create Project form this afternoon: There is a form section dedicated to setting the step size when the simulation type is selected as SedOneStepSimulation. However, the Simulators api does not seem to return any simulators with any algorithms that list this as a compatible simulation type. So, it never appears in the form and can never be selected.

Is this simulation type something that has been deprecated and is no longer needed? Or is it something that will be supported down the line?

**Comment by jonrkarr** [comment, marked as answer]
In short, ignore this class.

This is part of the SED-ML data model. I believe SedOneStepSimulation was intended to enable investigators to describe co-simulations where multiple models (e.g., a bacterium and a mammalian cell that it attacks) are simulated together.  However, I haven't seen anyone use this class. I also don't think SED-ML is a good way to describe such co-simulations. The semantics that need to be captured are richer than what can be captured with the current version of SED-ML. I think this class is something that the creators thought might be useful, but hasn't turned out to be helpful.

**Reply by jmdetloff** [reply]
Got it 👍 With that in mind should I just leave out the OneStep option from the simulation type dropdown, and the form section for setting step size, when updating the create project form?

**Reply by jonrkarr** [reply]
I don't have a strong feeling either way.

In the interest of being complete, the form could include the OneStep option. In that case, the form should have an input for the step size.

Alternatively, we could argue that showing a OneStep option isn't particular helpful because no simulation tool has the ability to execute OneStep simulation tasks. Furthermore, single step simulations can be achieved using UniformTimeCourse with (outputEndTime - initialTime) = stepSize and numberOfSteps=1.

**Reply by jmdetloff** [reply]
Alright, in that case I'll go ahead and remove this for now (barring any objections). Less code is a win for maintainability. It can always be re-added if a simulator is added with the capability.

**Reply by jonrkarr** [reply]
Sounds good. Plus, from the user's standpoint its probably best to keep it simpler by avoiding options that they can't fully utilize.
